### ISSUE TITLE
Warn admins about enabling both SAML and other authentication providers

### ISSUE THREAD

**Issue by nwalters512** [PrairieLearn/PrairieLearn#6164]
Because of the way we utilize both UIDs and UINs, there may be unexpected consequences if individuals can log in with both Google/Azure *and* SAML, as the former won't have the same UINs as the latter. We should warn admins if they try to enable other authentication providers alongside SAML.

See more discussion here: https://github.com/PrairieLearn/PrairieLearn/pull/6162#discussion_r955520180

**Comment by trombonekenny** [comment]
Yeah, I've never really been comfortable with us honoring the same UID coming from multiple providers (Shib and Gmail) as the same thing without some sort of user validation. Not only does it open up to potential abuse (could someone signup with another provider with the same credential/UIN as me and then use that to get in? Rare and might imply some bad IDP, but...) Also the attributes we get from the providers are different and might not be universally mappable to each other.

One solution might be to have a person entity that can be associated with multiple logins, and once those logins are validated as being part of the same person (email validation, or making them login as the first provider to validate adding the second provider to the person), then we can use any of the provider attributes we receive for the person even if they aren't logged in as one (UIN from Shib even when I'm logged in with, say, my GitHub account).

**Comment by nwalters512** [comment]
This was done in #13074.
